### PR TITLE
runner and detection

### PR DESCRIPTION
**Pull request by herjy**
This PR contains my original detection code, though I think @fred3m has a better, more modular one now that I haven't seen in the code. I also added a Runner class that runs scarlet with simplified input.

### CODE REVIEW

File: scarlet_extensions/Run/runner.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        catalog: dict
+            detection catalog that contains at least the position of the detected sources
+        bg_rms: float
+            background root mean square of the image
+        '''
+        self.lvl = lvl
+        self.wavelet = wavelet
+        catalog, self.bg_rms = makeCatalog(self._data, lvl, wavelet)
+        # Get the source coordinates from the HST catalog
+        self.pixels = np.stack((catalog['y'], catalog['x']), axis=1)
```
Comment (fred3m) [inline scarlet_extensions/Run/runner.py]: may `self.pixels` -> `self.pixel_coords`?

File: scarlet_extensions/Run/runner.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        ----------
+        datas: list of Data objects
+            the list of Data objects to run scarlet on
+        model_psf: 'numpy.ndarray' or `scarlet.PSF`
+            the target psf of the model
+        ra_dec: `array`
+            ra and dec positions of detected sources
+    """
+
+    def __init__(self, datas, model_psf, ra_dec = None):
```
Comment (fred3m) [inline scarlet_extensions/Run/runner.py]: `datas` -> `data`? That makes more sense and is the same as the private variable that you assign to it in the init.

File: scarlet_extensions/Run/runner.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+class Runner:
+    """ Class that lets scarlet run on a set of `Data` objects
+
+        This class makes running on batches of images easier.
+        It runs on multiple or single observations alike.
+
+        parameters
+        ----------
+        datas: list of Data objects
+            the list of Data objects to run scarlet on
```
Comment (fred3m) [inline scarlet_extensions/Run/runner.py]: You might want to make it clear what the expected format of the data object is, eg. that it's a compressed numpy file with keys (`images` = 3D image cube, etc).

File: scarlet_extensions/initialization/detection.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import numpy as np
+import collections
+import scarlet
+from scarlet.wavelet import mad_wavelet, Starlet
+
+# Class to provide compact input of instrument data and metadata
+class Data:
+    def __init__(self,images, wcs, psfs, channels):
+        self._images = images
+        self.wcss = wcs
```
Comment (fred3m) [inline scarlet_extensions/initialization/detection.py]: one `s` in `wcss`. This just looks confusing.

**Review by fred3m — CHANGES_REQUESTED** [review]
This looks good overall, I just have a few minor changes requested in the PR comments. In addition, I don't think that "Run" is the best name for the location for the runner. At the very least it should be "run" (lower case) but perhaps it would be better to call it "scripts" so that it's clear that it contains scripts that are meant to be executable in python.

**Comment by fred3m** [comment]
Once you've merged this PR I'll push my changes to the detection algorithm and ask you to review that.

**Comment by herjy** [comment]
Changes pushed.

**Review by fred3m — APPROVED** [review]

**Comment by fred3m** [comment]
Great, thanks Remy. Feel free to merge.

**Pull request merged**